### PR TITLE
Respect `scaling_reserved_ram` feature flag

### PR DESCRIPTION
**Pull request by nickrolfe**
The amount of RAM given to the CodeQL evaluator is the machine's total memory size, minus a reserved amount.

Currently, the reserved amount is fixed at 1 GB (or 1.5 GB on Windows). When the `scaling_reserved_ram` feature flag is enabled, we also add 2% of the total memory size to the reserved amount. This allows for the fact that the kernel will consume more RAM (e.g. for page tables) on machines with more physical RAM, so we should see fewer analyses getting killed by the OOM killer.

N.B. This is my first time working on the Action, or even writing TypeScript. I hope my uses of `async`/`await` make sense.

### Merge / deployment checklist

- [ ] Confirm this change is backwards compatible with existing workflows.
- [ ] Confirm the [readme](https://github.com/github/codeql-action/blob/main/README.md) has been updated if necessary.
- [ ] Confirm the [changelog](https://github.com/github/codeql-action/blob/main/CHANGELOG.md) has been updated if necessary.

### CODE REVIEW

**Review by henrymercer — COMMENTED** [review]
Looks good.  It's perhaps a little cautious, but we might want to add an integration test that runs the CodeQL Action with this feature enabled.  To do so, we could make a copy of `pr-checks/checks/multi-language-autodetect.yml`, enable the feature flag via the environment variable, and remove the language autodetection checks.

**Comment by henrymercer** [comment]
Oh, do add a changelog note too!  [The one from 24th May](https://github.com/github/codeql-action/blob/main/CHANGELOG.md#234---24-may-2023) is a good example.

File: src/util.ts [hunk 1]
```diff
@@ -161,9 +161,22 @@ export async function withTmpDir<T>(
...
+async function getSystemReservedMemoryMegaBytes(
+  totalMemoryMegaBytes: number,
+  codeql: CodeQL,
+  features: FeatureEnablement
+): Promise<number> {
   // Windows needs more memory for OS processes.
-  return 1024 * (process.platform === "win32" ? 1.5 : 1);
+  const fixedAmount = 1024 * (process.platform === "win32" ? 1.5 : 1);
+
+  if (await features.getValue(Feature.ScalingReservedRam, codeql)) {
```
[the remark this answers is not in the document]
Comment (nickrolfe) [reply src/util.ts:172]: Ah, I missed that that second parameter was optional. Thanks.

**Comment by nickrolfe** [comment]
I see the new integration test is running with `--ram=5769`, while the multi-language autodetection test runs with `--ram=5769`, so I think it's working.

**Review by henrymercer — APPROVED** [review]
Excellent!  (I think you meant the multi-language autodetection test runs with `--ram=5907`.)

**Pull request merged**